### PR TITLE
Np 45509 DataEntryUpdateHandler - handle all dao types

### PR DESCRIPTION
**Pull request by monaullah**
More functionality in `DataEntryUpdateHandler`:
- Publish messages for `ApprovalStatus` entries
- Publish messages on different topics based on `Candidate` `applicable` value
- Do nothing when entry types `Period` or `Note` are updated
- Do not fail for whole batch if processing of one record fails

See todo, will implement sending to DLQ on failure in next PR

Might need to read topic names from env variables when i create a sns topic in the template, looking into it

### CODE REVIEW

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DynamoDbUtils.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    static Map<String, AttributeValue> getImage(DynamodbStreamRecord record) {
+        var image = nonNull(record.getDynamodb().getNewImage())
+                        ? record.getDynamodb().getNewImage()
+                        : record.getDynamodb().getOldImage();
+        return mapToDynamoDbAttributeValue(image);
+    }
+
+    private static Map<String, AttributeValue> mapToDynamoDbAttributeValue(
+        Map<String, com.amazonaws.services.lambda.runtime.events.models.dynamodb.AttributeValue> image) {
```
Comment (torbjokv) [inline event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DynamoDbUtils.java:27]: probably exists some alternative to this sdk1 package, but well well lets save something fun for the future :-)

Comment (torbjokv) [reply event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DynamoDbUtils.java:27]: or maybe its not ready yet well well, then its all good

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DataEntryUpdateHandler.java [hunk 2]
```diff
@@ -22,12 +46,101 @@ public DataEntryUpdateHandler(NotificationClient<NviPublishMessageResponse> snsC
...
+            joiner.add(arg);
+        }
+        return joiner.toString();
+    }
+
+    private static boolean isUnknownOperationType(OperationType operationType) {
+        return OperationType.UNKNOWN_TO_SDK_VERSION.equals(operationType);
+    }
+
+    private void publishToTopic(DynamodbStreamRecord record) {
```
Comment (brinxmat) [inline event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DataEntryUpdateHandler.java:101]: Can you fix occurrences of the `record` keyword where it is used as a variable name?

File: event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DynamoDbUtils.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+
+public final class DynamoDbUtils {
+
+    private DynamoDbUtils() {
+    }
+
+    static Map<String, AttributeValue> getImage(DynamodbStreamRecord record) {
```
Comment (brinxmat) [inline event-handlers/src/main/java/no/sikt/nva/nvi/events/db/DynamoDbUtils.java:19]: `record`

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by torbjokv — APPROVED** [review]

**Pull request merged**